core/src/test/scala/org/apache/spark/scheduler/BlacklistTrackerSuite.scala

```diff
@@ -0,0 +1,162 @@
+      None,
+      None)
+
+  // Variable name can indicate basic information of taskInfo
+  // The format is "taskInfo_executorId_taskId_hostName"
+  val taskInfo_1_1_hostA = new TaskInfo(1L,1,1,0L,"1","hostA", TaskLocality.ANY, false)
+  val taskInfo_1_2_hostA = new TaskInfo(2L,1,1,0L,"1","hostA", TaskLocality.ANY, false)
+  val taskInfo_2_1_hostA = new TaskInfo(1L,1,1,0L,"2","hostA", TaskLocality.ANY, false)
+  val taskInfo_2_2_hostA = new TaskInfo(2L,1,1,0L,"2","hostA", TaskLocality.ANY, false)
+  val taskInfo_3_3_hostB = new TaskInfo(3L,1,1,0L,"3","hostB", TaskLocality.ANY, false)
+
+  test ("expireExecutorsInBlacklist works ") {
+    val conf = new SparkConf().setAppName("test").setMaster("local")
+      .set("spark.ui.enabled","false")
+      .set("spark.scheduler.blacklist.recoverPeriod", "2")
+    sc = new SparkContext(conf)
+    val scheduler = new TaskSchedulerImpl(sc, 1)
+
+    val tracker = new BlacklistTracker(conf)
+    tracker.start()
+    Thread.sleep(100)
+    tracker.updateFailureExecutors(taskInfo_1_1_hostA, FAILURE)
+    assert(tracker.executorBlacklist(scheduler, 1L) === Set("1"))
+
+    // sleep 2 second to wait for expireExecutorsInBlackList be triggered
+    Thread.sleep(2000)
+    assert(tracker.executorBlacklist(scheduler, 1L) === Set())
+
+    tracker.stop()
+  }
+
+  test("default strategy works as exceptation") {
+    val conf = new SparkConf().setAppName("test").setMaster("local")
+      .set("spark.ui.enabled","false")
+    sc = new SparkContext(conf)
+    val scheduler = new TaskSchedulerImpl(sc, 1)
+
+    val tracker = new BlacklistTracker(conf)
+    tracker.updateFailureExecutors(taskInfo_1_1_hostA, FAILURE)
+    tracker.updateFailureExecutors(taskInfo_2_1_hostA, FAILURE)
+    assert(tracker.executorBlacklist(scheduler, 1L) === Set("1","2"))
+    assert(tracker.executorBlacklist(scheduler, 2L) === Set())
+
+    tracker.updateFailureExecutors(taskInfo_1_1_hostA, Success)
+    assert(tracker.executorBlacklist(scheduler, 1L) === Set("2"))
+    assert(tracker.nodeBlacklist() === Set())
+
+    tracker.updateFailureExecutors(taskInfo_3_3_hostB, FAILURE)
+    assert(tracker.executorBlacklist(scheduler, 3L) === Set("3"))
+    assert(tracker.nodeBlacklist() === Set())
+
+    tracker.updateFailureExecutors(taskInfo_2_1_hostA, Success)
+    assert(tracker.executorBlacklist(scheduler, 1L) === Set())
+  }
+  
+  test ("strict strategy works as exception") {
+    val conf = new SparkConf().setAppName("test").setMaster("local")
+      .set("spark.ui.enabled","false")
+      .set("spark.scheduler.blacklist.strategy", "strict")
+    sc = new SparkContext(conf)
+    val scheduler = new TaskSchedulerImpl(sc, 1)
+
+    val tracker = new BlacklistTracker(conf)
+    tracker.updateFailureExecutors(taskInfo_1_1_hostA, FAILURE)
+    assert(tracker.executorBlacklist(scheduler, 1L) === Set("1"))
+    tracker.updateFailureExecutors(taskInfo_1_2_hostA, FAILURE)
+    assert(tracker.executorBlacklist(scheduler, 2L) === Set("1"))
```

Comment:
As the strict strategy/threshold strategy is taskID unrelated, executor 1 is in the blacklist for all tasks.